static_src/main.js

```diff
@@ -31,10 +31,18 @@ var updateModel = function() {
                 }
 
                 if (!_.findByKey(category.children, c[1])) {
-                    category.children.push({
-                        key: c[1],
-                        active: true,
-                    });
+                    if (state == null) {
+                        category.children.push({
+                            key: c[1],
+                            active: true,
+                        });
+                    } else {
+                        var stateSubcategory = _.findByKey(_.findByKey(state.categories, c[0]).children, c[1]);
+                        category.children.push({
+                            key: c[1],
+                            active: stateSubcategory? stateSubcategory.active: false,
```

Comment:
There should be spaces:

```suggestion
                            active: stateSubcategory ? stateSubcategory.active : false,
```

Reply:
Also if the category didn't exist before, I think the better behavior would be:

- If previously all categories were activated, also activate the new one
- If previously only some categories were activated, do not activate the new one.

On the other hand, it is likely that a new category has just been added by this user, so it would probably be helpful to activate that category. This would also be much less code. Then again, if the new category was added by another user, this would be confusing. What do you think?

Reply:
> * If previously all categories were activated, also activate the new one
> 
> * If previously only some categories were activated, do not activate the new one.

I like this behavior and will implement it.

Reply:
I am actually torn on this one.

Either the default depends on the previous state, in which the refactoring I proposed above (store active categories and available categories separately and only update available categories from the server) wouldn't work. It would also add complexity. And I am not even sure it is the better option from a user perspective.

Or we do the opposite: Always activate new categories and do the refactoring.

I actually ended up doing neither of these options: always activate new categories, but without the refactoring. You could call it "lazy" or "keeping all options on the table". I call it "good enough".